extensions/patterns/configure_ec2/playbooks/terminate_ec2_instance.yml

```diff
@@ -0,0 +1,24 @@
+---
+- name: Terminate EC2 instance
+  hosts: localhost
+  gather_facts: false
+  module_defaults:
+    group/aws:
+      aws_region: "{{ aws_region }}"
+  tasks:
+    - name: Set ec2_instance_create_delete role vars
+      ansible.builtin.set_fact:
+        ec2_instance_create_delete_operation: delete
+        ec2_instance_create_delete_instance_name: "{{ instance_name }}"
+        ec2_instance_create_delete_key_name: "{{ key_name | default(omit, true) }}"
+
+    - name: Delete EC2 instance
+      ansible.builtin.include_role:
+        name: cloud.aws_ops.ec2_instance_create_delete
```

Comment:
It turns out the `default(omit)` filter doesn't work when passing vars to a role, only to a module. [This is apparently the expected behavior](https://github.com/ansible/ansible/issues/22417). Thus, I had to use the `set_fact` module to set the `ec2_instance_create_delete_key_name` var so it would be omitted in `key_name` is not present.